sentry.conf.py

```diff
@@ -250,46 +251,52 @@
     # 'workers': 3,  # the number of web workers
 }
 
-###############
-# Mail Server #
-###############
-
 
-email = env('SENTRY_EMAIL_HOST') or (env('SMTP_PORT_25_TCP_ADDR') and 'smtp')
-if email:
-    SENTRY_OPTIONS['mail.backend'] = 'smtp'
-    SENTRY_OPTIONS['mail.host'] = email
-    SENTRY_OPTIONS['mail.password'] = env('SENTRY_EMAIL_PASSWORD') or ''
-    SENTRY_OPTIONS['mail.username'] = env('SENTRY_EMAIL_USER') or ''
-    SENTRY_OPTIONS['mail.port'] = int(env('SENTRY_EMAIL_PORT') or 25)
-    SENTRY_OPTIONS['mail.use-tls'] = env('SENTRY_EMAIL_USE_TLS', False)
-else:
-    SENTRY_OPTIONS['mail.backend'] = 'dummy'
-
-# The email address to send on behalf of
-SENTRY_OPTIONS['mail.from'] = env('SENTRY_SERVER_EMAIL') or 'root@localhost'
 
-# If you're using mailgun for inbound mail, set your API key and configure a
-# route to forward to /api/hooks/mailgun/inbound/
-SENTRY_OPTIONS['mail.mailgun-api-key'] = env('SENTRY_MAILGUN_API_KEY') or ''
+##########
+# Docker #
+##########
 
-# If you specify a MAILGUN_API_KEY, you definitely want EMAIL_REPLIES
-if SENTRY_OPTIONS['mail.mailgun-api-key']:
-    SENTRY_OPTIONS['mail.enable-replies'] = True
-else:
-    SENTRY_OPTIONS['mail.enable-replies'] = env('SENTRY_ENABLE_EMAIL_REPLIES', False)
+# Docker's environment configuration needs to happen
+# prior to anything that might rely on these values to
+# enable more "smart" configuration.
+
+ENV_CONFIG_MAPPING = {
+    'SENTRY_EMAIL_PASSWORD': 'mail.password',
+    'SENTRY_EMAIL_USER': 'mail.username',
+    'SENTRY_EMAIL_PORT': ('mail.port', int),
+    'SENTRY_EMAIL_USE_TLS': ('mail.use-tls', bool),
+    'SENTRY_EMAIL_HOST': 'mail.host',
+    'SENTRY_SERVER_EMAIL': 'mail.from',
+    'SENTRY_ENABLE_EMAIL_REPLIES': 'mail.enable-replies',
```

Comment:
I had to make this be a boolean:

```suggestion
    'SENTRY_ENABLE_EMAIL_REPLIES': ('mail.enable-replies', bool),
```